Cindy/Org CRUD API

cindyxzhang opened the pull request:
_Describe your feature here_
CRUD API for organizations at api/orgs

~ currently does not work and do not know why i am confusion.
## Related PRs

_Optional - any related PRs you're waiting on, or PRs that will conflict, etc_

## Migrations

_Optional - if you added anything to the database through migration(s)_

## Screenshots

_Required for visual changes_

CC: @kwfk

cindyxzhang committed:
feat(org api): started 4 API calls

cindyxzhang committed:
feat(api): started API

cindyxzhang committed:
feat(org api): rebasing

cindyxzhang committed:
feat(org pai): fixing lint

cindyxzhang committed:
feat(org api): added orgs API

currently does not work and i dont know why :')))

cindyxzhang committed:
feat(api works): org API works, dont get latest node yall

kwfk commented on pages/api/orgs/index.ts:
```diff
@@ -0,0 +1,108 @@
+  res: NextApiResponse
+): Promise<void> => {
+  if (req.method === 'POST') {
+    handleCREATE(req, res);
+  } else if (req.method === 'DELETE') {
+    handleDELETE(req, res);
+  } else if (req.method === 'PATCH') {
+    handlePATCH(req, res);
+  } else if (req.method === 'GET') {
+    handleGET(req, res);
```
let's change the names to use camel case so `handleCreate`

kwfk replied:
actually let's use HTTP verbs as the name as well so `handleGet`

kwfk commented on pages/api/orgs/index.ts:
```diff
@@ -0,0 +1,108 @@
+      });
+      res.json(newOrg);
+    } catch (err) {
+      console.log(err);
```
send a 500 error here

kwfk commented on pages/api/orgs/index.ts:
```diff
@@ -0,0 +1,108 @@
+    }
+  } else {
+    res
+      .status(500)
```
this should be a 400 error

kwfk commented on pages/api/orgs/index.ts:
```diff
@@ -0,0 +1,108 @@
+  } else {
+    res
+      .status(500)
+      .json({ statusCode: 500, message: 'Org name not passed in.' });
```
format the json as
```json
{
    error: {
         status: 400
         message: "..."
    }
}
```

kwfk replied:
@claalmve

kwfk committed:
feat(organization api): update organization API with more validation and error handling

Kept POST in index but moved GET, PATCH, DELETE to separate [id] route so that ID is in the route
instead of query params to be more inline with the page routes. Added schema validation with Joi.
Added error message utils.
